xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionXDOMService.java

```diff
@@ -86,20 +83,18 @@ private static boolean matchMentionMacro(Block block)
     }
 
     @Override
-    public Map<DocumentReference, List<String>> groupAnchorsByUserReference(List<MacroBlock> mentions,
-        WikiReference wikiReference)
+    public Map<MentionedActorReference, List<String>> groupAnchorsByUserReference(List<MacroBlock> mentions)
     {
-        Map<DocumentReference, List<String>> ret = new HashMap<>();
+        Map<MentionedActorReference, List<String>> ret = new HashMap<>();
         for (MacroBlock block : mentions) {
-            String serializedUserReference = block.getParameter(REFERENCE_PARAM_NAME);
+            String userReference = block.getParameter(REFERENCE_PARAM_NAME);
             // We are currently resolving to DocumentReference to allow us to support the mention of groups.
-            DocumentReference reference =
-                this.documentReferenceResolver.resolve(serializedUserReference, wikiReference);
-            String anchor = block.getParameter(ANCHORID_PARAM_NAME);
-            ret.merge(reference, new ArrayList<>(Collections.singletonList(anchor)), (l1, l2) -> {
-                l1.addAll(l2);
-                return l1;
-            });
+
+            ret.merge(new MentionedActorReference(userReference, block.getParameter(TYPE_PARAM_NAME)),
```

Comment:
Note that the type parameter is new, so you need to consider that this parameter might not exist for some old mentions: maybe it's handled later in the code?

Reply:
You are right, if the type is not defined, `MentionedActorReference` is initialized with a `null`, the `null` case is handled downstream.

Reply:
Wouldn't that be safer to check here that the param is null and to use a default "user" value type to put in the map in that case? Because right now you're creating a Map that is available in different APIs and it's rarely expected to have a null value as key.

Reply:
That might avoid some inconsistencies later, when the type is not defined, the "user" values is now used instead.